### ISSUE TITLE
4D Gabor

### ISSUE THREAD

**Issue by zhenjiangwu** [jolaem/Gabor3D#1]
Hello.
Do you think the Gabor filter can be extended to 4D?

**Comment by jolaem** [comment]
Hi,

Theoretically yes - you only need to add another axis to the equation.

But in practice it would be very difficult to visualise and understand it. What do you need 4D Gabor for?

-Jola

**Comment by zhenjiangwu** [comment]
Thank you for reply,
I'm learning how to code CNN by tensorflow and keras,and how to apply it for hyperspectral image classification.In one of my program,I have to use a 4D Gabor as the Convolution kernel to extract the spatial feature from the image，because I have extracted 4 kinds of features by PCA. However,I find the almost no scholar use Gabor for 3 or 4 dimensional operation,so I'm so confused and wonder if I'm wrong.Luckily,you told me the answer.Thanks a lot.
-zhenjiang

At 2019-10-21 18:21:15, "jolaem" <notifications@github.com> wrote:


Hi,

Theoretically yes - you only need to add another axis to the equation.

But in practice it would be very difficult to visualise and understand it. What do you need 4D Gabor for?

-Jola
